app/src/main/java/org/astraea/app/partitioner/Dispatcher.java

```diff
@@ -62,13 +102,70 @@ default void configure(Map<String, ?> configs) {
   @Override
   default int partition(
       String topic, Object key, byte[] keyBytes, Object value, byte[] valueBytes, Cluster cluster) {
-    return partition(
-        topic,
-        keyBytes == null ? new byte[0] : keyBytes,
-        valueBytes == null ? new byte[0] : valueBytes,
-        CLUSTER_CACHE.computeIfAbsent(cluster, ignored -> ClusterInfo.of(cluster)));
+    return interdependent.isInterdependent
+        ? interdependent.interdependentPartition(
+            this,
+            topic,
+            keyBytes,
+            valueBytes,
+            CLUSTER_CACHE.computeIfAbsent(cluster, ignored -> ClusterInfo.of(cluster)))
+        : partition(
+            topic,
+            keyBytes == null ? new byte[0] : keyBytes,
+            valueBytes == null ? new byte[0] : valueBytes,
+            CLUSTER_CACHE.computeIfAbsent(cluster, ignored -> ClusterInfo.of(cluster)));
   }
 
   @Override
   default void onNewBatch(String topic, Cluster cluster, int prevPartition) {}
+
+  private void begin() {
+    synchronized (interdependent) {
+      interdependent.isInterdependent = true;
+    }
+  }
+
+  private void end() {
+    synchronized (interdependent) {
+      interdependent.isInterdependent = false;
+      interdependent.targetPartition = -1;
+    }
+  }
+
+  class Interdependent {
+    private boolean isInterdependent = false;
+    private int targetPartition = -1;
```

Comment:
This will make all producers in the same JVM use "same" partition. For each producer, the `group result` should be independent. Please fix it or file issue to improvement this later

Reply:
我用`dispatcher`的hashcode作爲Key,以此來進行區分。已實做並添加相關test。